docs/specifications/tests/MethodsV2.md

```diff
@@ -0,0 +1,1021 @@
+
+[To top]
+
+
+## Method: Get out-of-bailiwick ip addresses (internal)
+
+### Method identifier
+**Get-OOB-IPs**
+
+### Objective
+
+Obtain the IP addresses of the out-of-bailiwick name servers for the
+given zone (child zone) and a given set of name server names. Extract
+addresses even if the resolution goes through CNAME. It is,
+however, not permitted for a NS record to point at a name that has a
+CNAME, but that test is covered by Test Case [DELEGATION05].
+
+This is an [internal method] that can be referred to by other methods in this
+document, but not by Test Case specifications.
+
+### Inputs
+
+* "Child Zone" - The name of the child zone.
+* "NS Set" - The name servers names as given by the calling method.
+* "Test Type" - The test type with values "undelegated test" or
+  "normal test".
+
+This method also inherits the inputs of method [Get-Undel-Data] if
+*Test Type* is "undelegated test".
+
+### Prerequisite
+
+None.
+
+### Test procedure
+
+1. If *NS Set* is empty then then output an empty set and exit these test
+   procedures.
+
+2. Create a set of name servers where each unique name server name in *NS Set*
+   is linked to an empty set of IP addresses ("Name Servers").
+
+3. If *Test Type* is "undelegated test", then do:
+   1. Fetch name server name and IP address or addresses using method
+      [Get-Undel-Data] ("Undelegated Data").
+   2. If the *Undelegated Data* set is empty or undefined, then output an
+      undefined set and exit these test procedures.
+
+3. For each name server name ("Name") in *NS Set* do:
+
+   1. If *Test Type* is "undelegated test" and if the *Name*
+      has IP address specification (IPv4 or IPv6) in *Undelegated Data*,
+      then:
+      1. Add the address or addresses to *Name Servers* for *Name*.
+      2. Go to next server name server name.
+
+   2. Create the following two [DNS queries][DNS Query]:
+      1. Query type A and *Name* as the query name and the RD flag set true
+         ("A Query").
+      2. Query type AAAA and *Name* as the query name and the RD flga set
+         true ("AAAA Query").
+
+   3. Send the two queries doing normal recursive lookup to a resolver.
+
+   4. If the [DNS Responses][DNS Response], if any, contains a list of A or AAAA
+      records (follow any CNAME chain) in the answer section then remember the IP
+      addresses for next step.
+
+   5. Collect all IP addresses for the *Name* and add the address or addresses to
+      *Name Servers* for that *Name* and go to next *Name*.
+
+4. Output the *Name Servers*.
+
+### Outputs
+
+* A set of name servers, where each unique name server name links to a possibly
+  empty set of its IP addresses:
+  * Non-empty set: The normal case.
+  * Empty set: No addresses were available, also a normal case.
+  * Undefined set: [Get-Undel-Data] returned an empty or undefined set.
+
+### Special procedural requirements
+
+None.
+
+### Dependencies
+
+This method depends on methdo [Get-Undel-Data].
```

Comment:
methdo -> Method

Reply:
Fixed typo. So what you suggest is that everytime there is `method [LINK-2-METHOD]` it should be "Method" instead of "method"?

Reply:
Updated.